### PR TITLE
Allow null set on unevaluated property value

### PR DESCRIPTION
**Pull request by adamint**
Similar to evaluated property values, this allows null-set unevaluated property values. This does not preclude an empty unevaluated value from being set.
This is for #6920 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8083)

### CODE REVIEW

**Review by tmeschter — CHANGES_REQUESTED** [review]
This is going to cause problems; see https://github.com/dotnet/project-system/issues/6920#issuecomment-1101650588.

**Comment by adamint** [comment]
@tmeschter is this the way it should be done?

File: docs/repo/property-pages/back-end-architecture.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ The Project Query API treats its use of brokered services as an implementation d
 
 Internally, the Project Query API handles serializing the query, sending it across the network, deserializing it, and coordinating the actions of an extensible set of data providers. These data providers are responsible for filling in the requested properties on various entities, creating child entities from their parents, and handling various "actions" that update state in some way. For example, one provider may be responsible for creating project entities representing all the projects in the solution, while a different provider is responsible for filling in the project's name, path, GUID, language, etc. There may be multiple providers that handle the same relationship from a parent to its children or the same action. Among other things, the Project Query API implementation glues together the results of all of these providers to create the final query result.
 
+Please note that internally, setting either a property's evaluated or unevaluated value to null will call `IProject.DeleteAsync` to delete the property.
```
Comment (tmeschter) [inline docs/repo/property-pages/back-end-architecture.md:37]: I think this information should go in the comments for the relevant types in the CPS system. This is, fundamentally, a change in the semantics of the Project Query API itself (though we currently have the only implementation of that part of the API).

**Review by tmeschter — APPROVED** [review]
Yeah, that's more what I had in mind.

File: docs/repo/property-pages/back-end-architecture.md (hunk 1 above)
Comment (adamint) [reply docs/repo/property-pages/back-end-architecture.md:37]: I will make sure to document this. Taking on #6921

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/ProjectSetUnevaluatedUIPropertyValueAction.cs [hunk 2]
```diff
@@ -25,9 +25,9 @@ public ProjectSetUnevaluatedUIPropertyValueAction(SetUnevaluatedUIPropertyValue
 
         protected override Task SetValueAsync(IProperty property)
         {
-            if (property is IEvaluatedProperty evaluatedProperty)
+            if (_parameter.Value == null)
             {
-                return evaluatedProperty.SetUnevaluatedValueAsync(_parameter.Value ?? string.Empty);
+                return property.DeleteAsync();
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/ProjectSetUnevaluatedUIPropertyValueAction.cs:30]: Why don't we need to call `SetUnevaluatedValueAsync` here anymore?

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/ProjectSetUnevaluatedUIPropertyValueAction.cs:30]: Ah, I assume this is https://github.com/dotnet/project-system/issues/6920#issuecomment-1101569461

**Pull request merged**